dynamiqs/qarrays/sparse_qarray.py

```diff
@@ -70,6 +71,15 @@ def norm(self) -> Array:
     def unit(self) -> SparseQArray:
         return SparseQArray(self.diags / self.norm(), self.offsets, self.dims)
 
+    def isket(self) -> bool:
+        return isket(self.to_dense())
+
+    def isbra(self) -> bool:
+        return isbra(self.to_dense())
+
+    def isdm(self) -> bool:
+        return isdm(self.to_dense())
```

Comment:
These don't need a call to `.to_dense()` because the functions just check the object shape. This currently won't work because of the call to `jnp.asarray` but it's fixed in https://github.com/dynamiqs/dynamiqs/pull/623.

Reply:
Yes, no need to implement them.